Source/WebCore/platform/graphics/gtk/GdkSkiaUtilities.cpp

```diff
@@ -73,6 +73,17 @@ RefPtr<cairo_surface_t> skiaImageToCairoSurface(SkImage& image)
 }
 #endif
 
+GRefPtr<GdkPixbuf> skiaImageToGdkPixbuf(SkImage& image)
+{
+    SkPixmap pixmap;
+    if (!image.peekPixels(&pixmap))
+        return { };
+
+    return adoptGRef(gdk_pixbuf_new_from_data(pixmap.addr8(), GDK_COLORSPACE_RGB, !pixmap.computeIsOpaque(), pixmap.info().bytesPerPixel() * 8, pixmap.width(), pixmap.height(), pixmap.rowBytes(), [](guchar*, void *data) {
```

Comment:
According to the [documentation](https://docs.gtk.org/gdk-pixbuf/ctor.Pixbuf.new_from_data.html), `has_alpha` signifies “Whether the data has an opacity channel” but Skia's `::computeIsOpaque()` returns “true if all pixels are opaque”. I think this should use `SkPixmap::isOpaque()` which checks whether the image has alpha or not, regardless of whether any pixel is itself opaque, and this seems to match the expectation of the `gdk_pixbuf_new_from_data()` function better.

```suggestion
    return adoptGRef(gdk_pixbuf_new_from_data(pixmap.addr8(), GDK_COLORSPACE_RGB, !pixmap.isOpaque(), pixmap.info().bytesPerPixel() * 8, pixmap.width(), pixmap.height(), pixmap.rowBytes(), [](guchar*, void *data) {
```

Also, a couple of notes more:

- Using `SkPixmap::data8()` asserts that `::bytesPerPixel() == 1` so it will fail for anything that is not 8 bits per component if Skia is ever built with assertions enabled. We should use `::addr(0, 0)` if we want to support images with more than 8bpc.
- Alternatively, we may want to support only a fixed set of formats, and convert to `RGB{A,X}_8888` otherwise.

Reply:
For now ill just assert 8bit since GTK4 only supports that atm.

Reply:
> For now ill just assert 8bit since GTK4 only supports that atm.

Sounds fine by me. Maybe add a FIXME/TODO stating that we might want to consider converting the incoming image to to 8bpc (`RGB{A,X}_8888`) if needed.